adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -3051,4 +3176,27 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
       loadParquetFragments(pathName, optPredicate = optPredicate, optProjection = optProjection)
     }
   }
+
+  /**
+   * Return true if the specified path of Parquet + Avro files is partitioned.
+   *
+   * @param pathName Path in which to look for partitioned flag.
+   * @return Return true if the specified path of Parquet + Avro files is partitioned.
+   *
+   * If a glob is used, all directories within the blog must be partitioned, and must have been saved
+   * using the same partitioned bin size.  Behavior is undefined if this requirement is not met.
+   */
+  def getPartitionedBinSize(pathName: String): Int = {
+
+    val partitionSize = try {
+      val f = getFsAndFilesWithFilter(pathName, new FileFilter("_isPartitionedByStartPos")).head
+      f.getFileSystem(sc.hadoopConfiguration).open(f).readInt()
```

Comment:
You need to close the stream after you open it. E.g.:

```
val is = f.getFileSystem(sc.hadoopConfiguration).open(f)
val bucketSize = is.readInt()
is.close()
```

Additionally, I'd prefer that we enforced that all partitioned files had the same bin size. E.g.,

```
val partitionSizes = getFsAndFilesWithFilter(pathName, new FileFilter("_isPartitionedByStartPos")).map(f => {
    val is = f.getFileSystem(sc.hadoopConfiguration).open(f)
    val partitionSize = is.readInt()
    is.close()
    partitionSize
  }).toSet

require(partitionSizes.nonEmpty, "Input Parquet files (%s) are not partitioned.".format(pathName))
require(partitionSizes.size == 1, "Found multiple partition sizes (%s).".format(partitionSizes.mkString(", "))

partitionSizes.head
```

You shouldn't ever actually get a `FileNotFoundException`; if there's no `_isPartitionedByStartPos` files, then `getFsAndFilesWithFilter` shouldn't return any files.

Reply:
Done